hazelcast/src/main/java/com/hazelcast/config/Config.java

```diff
@@ -207,32 +214,254 @@ public Config(String instanceName) {
      * @return Config created from a file when exists, otherwise default.
      */
     public static Config load() {
-        return new ExternalConfigurationOverride().overwriteMemberConfig(loadFromFile());
+        return new ExternalConfigurationOverride().overwriteMemberConfig(load(System.getProperties()));
+    }
+
+    private static Config load(Properties properties) {
+        return loadFromFile(properties);
     }
 
-    private static Config loadFromFile() {
+    private static Config loadFromFile(Properties properties) {
         validateSuffixInSystemProperty(SYSPROP_MEMBER_CONFIG);
 
         XmlConfigLocator xmlConfigLocator = new XmlConfigLocator();
         YamlConfigLocator yamlConfigLocator = new YamlConfigLocator();
 
         if (xmlConfigLocator.locateFromSystemProperty()) {
             // 1. Try loading XML config from the configuration provided in system property
-            return new XmlConfigBuilder(xmlConfigLocator).build();
+            return new XmlConfigBuilder(xmlConfigLocator).setProperties(properties).build();
         } else if (yamlConfigLocator.locateFromSystemProperty()) {
             // 2. Try loading YAML config from the configuration provided in system property
-            return new YamlConfigBuilder(yamlConfigLocator).build();
+            return new YamlConfigBuilder(yamlConfigLocator).setProperties(properties).build();
         } else if (xmlConfigLocator.locateInWorkDirOrOnClasspath()) {
             // 3. Try loading XML config from the working directory or from the classpath
-            return new XmlConfigBuilder(xmlConfigLocator).build();
+            return new XmlConfigBuilder(xmlConfigLocator).setProperties(properties).build();
         } else if (yamlConfigLocator.locateInWorkDirOrOnClasspath()) {
             // 4. Try loading YAML config from the working directory or from the classpath
-            return new YamlConfigBuilder(yamlConfigLocator).build();
+            return new YamlConfigBuilder(yamlConfigLocator).setProperties(properties).build();
         } else {
             // 5. Loading the default XML configuration file
             xmlConfigLocator.locateDefault();
-            return new XmlConfigBuilder(xmlConfigLocator).build();
+            return new XmlConfigBuilder(xmlConfigLocator).setProperties(properties).build();
+        }
+    }
+
+    /**
+     * Loads Config using the default {@link #load() lookup mechanism} to locate the configuration
+     * file, except it does not override configuration properties using the environment variables
+     * or system properties.
+     *
+     * @return Config created from a file when exists, otherwise default.
+     */
+    public static Config loadDefault() {
+        return load(System.getProperties());
```

Comment:
Nicely done! 🏅 

Can you just make sure that all variants of the helper methods use `ExternalConfigurationOverride().overwriteMemberConfig()`, as it's a really useful feature that allows user to override most config without modifying XML or YAML files (https://github.com/hazelcast/hazelcast/pull/17348)?

Reply:
thanks :) done